include/fiction/io/print_layout.hpp

```diff
@@ -343,41 +353,147 @@ void print_charge_layout(std::ostream& os, const Lyt& lyt, const bool cs_color =
     static_assert(is_cell_level_layout_v<Lyt>, "Lyt is not a cell-level layout");
     static_assert(has_sidb_technology_v<Lyt>, "Lyt is not an SiDB layout");
     static_assert(has_siqad_coord_v<Lyt>, "Lyt is not based on SiQAD coordinates");
-    static_assert(has_get_charge_state_v<Lyt>, "Lyt does not implement the get_charge_state function");
+    static_assert(has_get_charge_state_v<Lyt> | has_get_sidb_defect_v<Lyt>,
+                  "Lyt does not implement the get_charge_state function");
 
     // empty layout
     if (lyt.is_empty())
     {
-        os << "[i] empty layout" << std::endl;
-        return;
+        if constexpr (has_get_sidb_defect_v<Lyt>)
+        {
+            if (lyt.num_defects() == 0)
+            {
+                os << "[i] empty layout" << std::endl;
+                return;
+            }
+        }
```

Comment:
This seems odd to me. Why would you not print a layout that doesn't have defects?

Reply:
Because then it is empty and does not have defects.

Reply:
Ah, it didn't show me the outside `if` statement in the diff view. Understood. All good!